kafkaesk/consumer.py

```diff
@@ -140,28 +145,30 @@ class BatchConsumer(aiokafka.ConsumerRebalanceListener):
 
     def __init__(
         self,
-        stream_id: str,
-        group_id: str,
-        coro: typing.Callable,
+        subscription: Subscription,
         app: "Application",
         event_handlers: typing.Optional[typing.Dict[str, typing.List[typing.Callable]]] = None,
-        concurrency: int = 1,
-        timeout_seconds: float = None,
         auto_commit: bool = True,
     ):
         self._initialized = False
-        self.stream_id = stream_id
-        self.group_id = group_id
-        self._coro = coro
+        self.stream_id = subscription.consumer_id
+        self.group_id = subscription.group
+        self._coro = subscription.func
         self._event_handlers = event_handlers or {}
-        self._concurrency = concurrency
-        self._timeout = timeout_seconds
+        self._concurrency = subscription.concurrency or 1
+        self._timeout = subscription.timeout
+        self._subscription = subscription
         self._close = None
         self._app = app
         self._last_commit = 0
         self._auto_commit = auto_commit
         self._tp: typing.Dict[aiokafka.TopicPartition, int] = {}
 
+        # We accept either pattern or a list of topics, also we might accept a single topic
+        # to keep compatibility with older API
+        self.pattern = subscription.pattern
+        self.topics = subscription.topics
```

Comment:
if both `pattern` and `topics` are provided, should an error be raised?

Reply:
yeah, good point, maybe we can raise a `AssertionError `?